Tensile/KernelWriterAssembly.py

```diff
@@ -4760,28 +4888,150 @@ def lraFinalOffset(self, kernel, tP):
+            kernel["DepthU"], \
+            vgpr(tP["gpr"]["lro"]), \
+            "")
+        if (kernel["LdsBlockSizePerPad"]//(kernel["DepthU"]*tP["bpe"]) > 1):
+          #Add pad for evry LdsBLockSizePerPad
+          kStr += inst("v_lshrrev_b32", \
+              vgpr(tmpVgprPadoffset), \
+              log2(kernel["LdsBlockSizePerPad"]//(kernel["DepthU"]*tP["bpe"])), \
+              vgpr(tP["gpr"]["lro"]), \
+              "")
+          kStr += inst("v_mul_lo_u32", \
+              vgpr(tmpVgprPadoffset), \
+              (kernel["LdsPad%s"%tc]*tP["bpe"]), \
+              vgpr(tmpVgprPadoffset), \
+              "")
+        else:
+          kStr += inst("v_mul_lo_u32", \
+              vgpr(tmpVgprPadoffset), \
+              (kernel["LdsPad%s"%tc]*tP["bpe"]), \
+              vgpr(tP["gpr"]["lro"]), \
+              "")
+        kStr += inst("v_and_b32", \
+            vgpr(tmpVgpr), \
+            hex(globalParameters["WavefrontWidth"]-1), \
+            vgpr("Serial"), \
+            "")
+        if kernel["MatrixInstB"] == 1:
+            kStr += inst("v_lshrrev_b32", \
+                vgpr(tmpVgpr), \
+                log2(kernel("MatrixInstM")), \
+                vgpr(tmpVgpr), \
+                "")
+        else:
+            kStr += inst("v_lshrrev_b32", \
+                vgpr(tmpVgpr), \
+                log2(globalParameters["WavefrontWidth"]), \
+                vgpr(tmpVgpr), \
+                "")
+        if (kernel["ProblemType"]["DataType"].isHalf()):
+          kStr += inst("v_lshlrev_b32", \
+              vgpr(tmpVgpr), \
+              log2(2), \
+              vgpr(tmpVgpr), \
+              "")
+        #kStr += inst("v_add_u32", \
+        #    vgpr(tmpVgpraddr), \
+        #    vgpr(tmpVgpr), \
+        #    vgpr(tmpVgpraddr), \
+        #    "")
+        #kStr += inst("v_lshlrev_b32", \
+        #    vgpr("LocalReadAddr%s"%tc), \
+        #    log2(tP["bpe"]), \
+        #    vgpr(tmpVgpraddr), \
+        #    "")
+        kStr += inst("_v_add_lshl_u32", \
+            vgpr("LocalReadAddr%s"%tc), \
+            vgpr(tmpVgpraddr), \
+            vgpr(tmpVgpr), \
+            hex(log2(tP["bpe"])), \
+            "")
+	#Add Lds PAD logic
+        kStr += inst("v_add_u32", \
+            vgpr("LocalReadAddr%s"%tc), \
+            vgpr(tmpVgprPadoffset), \
+            vgpr("LocalReadAddr%s"%tc), \
+            "")
+
+        if tc == "B": # For BBlocks, A and B use this case
+          #add perWave LDS base offset
+          if tP["grcg"]:
+            if tP["grcv"]:
+              divisorName = tP["lvc"]
+            else:
+              # Fractional load use the more accurate lsc, multiply by VW later
+              divisorName = tP["lsc"]
+          else:
+            if tP["grcv"]:
+              divisorName = tP["lsp"]
+            else:
+              divisorName = tP["lvp"]
+          numPerpElementsPerLoad = kernel[divisorName]
+          numPerpElementsPerWave = tP["nrp"]*numPerpElementsPerLoad
```

Comment:
should be 
numPerpElementsPerWave = tP["nrp"]* globalParameters["WavefrontWidth"] // numPerpElementsPerLoad

Reply:
fixed.